ethcore/src/json_tests/chain.rs

```diff
@@ -108,24 +118,30 @@ pub fn json_chain_test<H: FnMut(&str, HookType)>(json_data: &[u8], start_stop_ho
 
 		if !fail {
 			flushln!("ok");
+		} else {
+			flushln!("fail");
 		}
 
 		start_stop_hook(&name, HookType::OnStop);
 	}
 
-	println!("!!! {:?} tests from failed.", failed.len());
+	if failed.len() > 0 {
+		println!("!!! {:?} tests failed.", failed.len());
+	}
 	failed
 }
 
 #[cfg(test)]
 mod block_tests {
+	use std::path::Path;
+
 	use super::json_chain_test;
 	use json_tests::HookType;
 
-	fn do_json_test<H: FnMut(&str, HookType)>(json_data: &[u8], h: &mut H) -> Vec<String> {
-		json_chain_test(json_data, h)
+	fn do_json_test<H: FnMut(&str, HookType)>(path: &Path, json_data: &[u8], h: &mut H) -> Vec<String> {
+		json_chain_test(path, json_data, h)
 	}
-
+	//todo[dvdplm] do these tests match all folders in `res/` or are there tests we're missing?
```

Comment:
Yes, see [test! macro](https://github.com/paritytech/parity-ethereum/blob/master/ethcore/src/json_tests/test_common.rs#L107-#L112)

Reply:
> if all folders are `declared`

Right, that's what the todo is about.

Reply:
I actually know all folders are not covered. I have **no** idea if they should be though, e.g. `LegacyTests`.

Reply:
Ok, let's keep it out of this PR but we could probably discuss it in a separate issue. I don't know TBH.